src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2303,22 +2304,42 @@ private bool ConsumeCommentMultiSegment()
             }
 
             byte marker = localBuffer[0];
-
-            if (marker == JsonConstants.Slash)
+            if (marker != JsonConstants.Slash && marker != JsonConstants.Asterisk)
             {
-                return ConsumeSingleLineCommentMultiSegment(localBuffer.Slice(1), leftOver, start, _consumed);
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
             }
 
-            if (marker != JsonConstants.Asterisk)
+            localBuffer = localBuffer.Slice(1);
+            if (localBuffer.Length == 0)
             {
-                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, marker);
+                if (IsLastSpan)
+                {
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedComment);
+                }
+                if (!GetNextSpan())
+                {
+                    _totalConsumed = previousTotalConsumed;
+                    return false;
+                }
+                _totalConsumed++;
+                _bytePositionInLine++;
+                localBuffer = _buffer;
+                leftOver = 0;
             }
 
-            return ConsumeMultiLineCommentMultiSegment(localBuffer.Slice(1), leftOver, start, _consumed);
+            SequencePosition start = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed + leftOver);
```

Comment:
Would that not be a case for any other accumulation into `_consumed` for the number of bytes consumed within that span? Say number of bytes accumulated after consuming int / int64 or a string, etc. `_consumed` is linked with the _current span_ that is set in by the `GetNextSpan` call. So, as long as width of `_consumed` is sufficiently large to hold total number of bytes in the span, there should not be any overflow. Mainly because `_consumed` is updated only after the token has been parsed and its bytes consumed, else it lags as in above case.